src/validators/index.js

```diff
@@ -0,0 +1,42 @@
+const PropTypes = require('prop-types')
+
+//The guid validator checks the length of the prop since the guid format is consistent.
+//I can't think of a situation where we're trying to pass through a incorrect prop
+//that's a string of 36 characters.
+
+function chainedGuidLengthChecker(isRequired) {
+  return function(props, propName, componentName) {
+    const guid = props[propName] || ''
+
+    if (!guid && isRequired) {
+      return new Error(
+        `${propName} was empty. This is a required prop for ${componentName}`,
+      )
+    } else if (guid.length !== 36) {
+      return new Error(
+        `${props[propName]} supplied to ${componentName}, expected a guid.`,
+      )
+    }
+  }
+}
+
+let guid = chainedGuidLengthChecker(false)
```

Comment:
This should be a `const`

In JS `let` should be favored if you are going to overwrite or mutate the value...

```
let a = 42
// blah blah blah
a = 43
```

And `const` should be used for basically everything else, including when you're mutating that object's attributes:

```
const fortyTwo = 42
fortyTwo.three = 3 // stupid, i know
```